aspects/aspects.go

```diff
@@ -306,57 +328,144 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 		return a.bundle.schema.Validate(data)
 	}
 
-	return &NotFoundError{
-		Account:    a.bundle.Account,
-		BundleName: a.bundle.Name,
-		Aspect:     a.Name,
-		Field:      request,
-		Cause:      "field not found",
+	return notFoundErrorFrom(a, request, "no matching rule")
+}
+
+// Get returns the aspect value identified by the request. If either the named
+// aspect or the corresponding value can't be found, a NotFoundError is returned.
+func (a *Aspect) Get(databag DataBag, request string, value *interface{}) error {
+	matches, err := a.matchGetRequest(request)
+	if err != nil {
+		return err
+	}
+
+	var merged interface{}
+	for _, match := range matches {
+		var val interface{}
+		if err := databag.Get(match.storagePath, &val); err != nil {
+			if errors.Is(err, PathNotFoundError("")) {
+				continue
+			}
+			return err
+		}
+
+		// build namespace around result
+		namespace := match.namespace
+		for i := len(namespace) - 1; i >= 0; i-- {
+			val = map[string]interface{}{namespace[i]: val}
+		}
+
+		// merge result with results from other matching rules
+		merged, err = merge(merged, val)
+		if err != nil {
+			return err
+		}
+	}
+
+	if merged == nil {
+		return notFoundErrorFrom(a, request, "no stored value for matching rules")
+	}
+
+	// the top level maps the request to the remaining namespace
+	*value = map[string]interface{}{request: merged}
+	return nil
+}
+
+func merge(old, new interface{}) (interface{}, error) {
+	if old == nil {
+		return new, nil
+	} else if new == nil {
+		return old, nil
```

Comment:
this seems not to be reached by tests? also shouldn't we return nil for the value instead of old?

Reply:
Actually I don't think this can happen, I think it's just outdated. Null is not a value we can store so we can't get it at the top level or as a leaf